### PR TITLE
tig: 2.2 -> 2.2.1

### PR DESCRIPTION
**Pull request by sigma**
###### Motivation for this change

Catch up with upstream.

Also move to different project URLs, as requested in
https://github.com/jonas/tig/releases/tag/tig-2.2.1

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by dmalikov — APPROVED** [review]

File: pkgs/applications/version-management/git-and-tools/tig/default.nix [hunk 1]
```diff
@@ -3,11 +3,11 @@
...
+  name = "tig-2.2.1";
 
   src = fetchurl {
-    url = "http://jonas.nitro.dk/tig/releases/${name}.tar.gz";
-    sha256 = "0k3m894vfkgkj7xbr0j6ph91351dl6id5f0hk2ksjp5lmg9i6llg";
+    url = "https://github.com/jonas/tig/releases/download/${name}/${name}.tar.gz";
+    sha256 = "0wgsqdly3jd9c7b0ibb5vwsv4js19c5hi0698nf1fnfyjq40hj0b";
   };
 
   buildInputs = [ ncurses asciidoc xmlto docbook_xsl readline git makeWrapper ]
```
Comment (ndowens) [inline pkgs/applications/version-management/git-and-tools/tig/default.nix]: makeWrapper belongs in nativeBuildInputs

File: pkgs/applications/version-management/git-and-tools/tig/default.nix [hunk 2]
```diff
@@ -3,11 +3,11 @@
 }:
 
 stdenv.mkDerivation rec {
-  name = "tig-2.2";
+  name = "tig-2.2.1";
 
   src = fetchurl {
-    url = "http://jonas.nitro.dk/tig/releases/${name}.tar.gz";
```
Comment (ndowens) [inline pkgs/applications/version-management/git-and-tools/tig/default.nix]: Have you tried fetchFromGitHub, as it is now the preferred method when possible. When you do it, rev is most likely gonna be:
rev = "v${version}"; sometimes it is rev = version; according how the upstream lists their version

**Comment by sigma** [comment]
@ndowens that was a little bit more involved than I thought (turns out the .tar.gz was a pre-configured version of the sources), but I think it should be ok now.

**Comment by sigma** [comment]
hmm, not sure what's going on with the OSX build (which works in my own setup)

**Comment by sigma** [comment]
ok, I'm finally able to reproduce, by uninstalling the `docbook-xsl` homebrew package... something is seriously wrong right there.

**Comment by sigma** [comment]
and what was wrong was that tig's Makefile kinda goes out of its way to make it's as impure as possible :)
Interesting learning experience at least !

**Comment by peterhoeg** [comment]
Instead of adding automake/autoconf and manually running ./autogen.sh, you can *probably* add ```autoreconfHook``` to ```nativeBuildInputs```. That should take care of it automatically.

**Comment by ndowens** [comment]
I tried autoreconfHook but it didn't work

**Comment by sigma** [comment]
@peterhoeg thanks, that seems to do the trick. @ndowens am I missing something?

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**